controllers/vaultdynamicsecret_controller.go

```diff
@@ -205,9 +232,8 @@ func (r *VaultDynamicSecretReconciler) Reconcile(ctx context.Context, req ctrl.R
 		_ = helpers.HandleRolloutRestarts(ctx, r.Client, o, r.Recorder)
 	}
 
-	if !r.isRenewableLease(secretLease, o) {
-		logger.Info("Secret lease is not renewable, will not requeue for renewal",
-			"lease_id", secretLease.ID)
+	if (!r.isRenewableLease(secretLease, o) && !o.Spec.AllowStaticCreds) || horizon.Seconds() == 0 {
+		// no need to requeue
```

Comment:
I believe this will result in an Event of type Warning from `isRenewableLease()` being emitted even though we are not requeueing or returning an error, that seems a little off to me?

Reply:
Updated in e530736f78ad9e7a296cc53d2b0b347a2bfd1dce